### ISSUE TITLE
Workflow Orchestration Stuck in Running state for hours

### ISSUE THREAD

**Issue by gorillapower** [Azure/azure-functions-durable-extension#2072]
### Discussed in https://github.com/Azure/azure-functions-durable-extension/discussions/2062

<div type='discussions-op-text'>

<sup>Originally posted by **gorillapower** February  1, 2022</sup>
One of our Workflows occasionally never complete. We are seeing this once or twice a day currently (2 out of every +-50 runs)

The current UTC time is 10:32 so the function has been 'stuck' in a running state since 02:56 (+7 hours)
Can anyone guide me as to how to debug this issue? I dont see any messages in the control or worker queues.

![image](https://user-images.githubusercontent.com/8879065/151952908-6311d932-f216-454b-b79e-b269bd0ea718.png)

InstanceId: 845a0f2b-d3e0-4d26-9237-2c7fb3c73271 (West Europe) *Please note there will be two functions with the same instanceId in our environment, this is by design.
Plan: Consumption Plan
Runtime: nodejs
FUNCTIONS_EXTENSION_VERSION: ~3
FUNCTIONS_WORKER_RUNTIME_VERSION : 12

package.json
```{
  "dependencies": {
    "azure-storage": "^2.10.1",
    "durable-functions": "1.4.3",
    "moment": "^2.22.2",
    "uuid": "^3.3.2",
    "vm2": "3.6.3"
  }
} 
```
</div>

**Comment by cgillum** [comment]
Oddly, I noticed there were two different orchestration instances with this exact same instance ID during this time and Azure region (West EU), but for different task hubs. Is it correct that your task hub name starts with "spu"? The other task hub name starts with "efs".

**Comment by cgillum** [comment]
BTW, it looks like you're running on a really old version of the Durable extension ([v1.8.6](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.6)). Any chance I could convince you to upgrade to the latest version of extension bundles? Not only will that ensure you have the latest reliability fixes, but it will improve the logging.

It would be great to update to the latest version of the [durable-functions NPM package](https://www.npmjs.com/package/durable-functions) as well, which is currently [v1.5.4](https://github.com/Azure/azure-functions-durable-js/releases/tag/1.5.4).

**Comment by cgillum** [comment]
Also, I took a look at the [Durable Task Azure Storage logs](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-diagnostics?tabs=csharp#durable-task-framework-logging) for your app. The Durable Task Framework seems to think no new work was scheduled after your first activity completed. This leads me to believe it's either a code issue in your orchestrator logic (maybe some non-deterministic code?) or maybe an issue with the version of the JavaScript SDK your app is using.

BTW, upgrading your extension version will make it possible for you to collect these detailed logs as well.

**Comment by gorillapower** [comment]
@cgillum thanks for the response. Yes the two orchestrations with the same IDs but in different taskhubs is by design, we use that so that we can track the two orchestrations more easily via the HTTP APIs. I remember why I did this, because when in certain states, there is no SuborchestrationId in the `historyEvents[]`. See my issue here https://github.com/Azure/azure-functions-durable-extension/issues/854

Mmm, the orchestration code is very simple, it just calls the activity and returns the response from the activity.

Orchestration
```
const df = require("durable-functions");

module.exports = df.orchestrator(function* (context) {
    var input = context.df.getInput();
    var output = yield context.df.callActivity("UserFunction", input);
    return output;
});
```

But now that you mention it, I have noticed that a non-deterministic error does occur sometimes. Is it possible that it could be from the activitiy function code? That code is userdefined. UPDATE: I found an instance where we saw the non-deterministic error. InstanceID: 42fec86a-3661-41cb-aad3-8652a8bfc8ff

Yes I will upgrade the libraries to the latest versions, that should not be a problem.

Also worth noting, we have not seen an issue since I reduced the concurrent `maxConcurrentActivityFunctions` and `maxConcurrentOrchestratorFunctions` from 10 to 2.

Perhaps the original symptom is a non-deterministic error like you mentioned, but there was no log for it? I will try upgrade the libraries and see if we can reproduce the issue and with the improved logging, we may get to the root cause.

Thanks for your support
